core/src/test/java/org/apache/accumulo/core/data/KeyBuilderTest.java

```diff
@@ -0,0 +1,282 @@
+    Key keyExpected = new Key(rowText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyText() {
+    Key keyBuilt = Key.builder().row(rowText).family(familyText).build();
+    Key keyExpected = new Key(rowText, familyText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyQualifierText() {
+    Key keyBuilt = Key.builder().row(rowText).family(familyText).qualifier(qualifierText).build();
+    Key keyExpected = new Key(rowText, familyText, qualifierText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyQualifierVisibilityText() {
+    Key keyBuilt  = Key.builder().row(rowText).family(familyText).qualifier(qualifierText).visibility(visibilityText).build();
+    Key keyExpected = new Key(rowText, familyText, qualifierText, visibilityText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyQualifierVisibilityTimestampText() {
+    Key keyBuilt = Key.builder().row(rowText).family(familyText).qualifier(qualifierText).visibility(visibilityText).timestamp(1L).build();
+    Key keyExpected = new Key(rowText, familyText, qualifierText, visibilityText, 1L);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyQualifierVisibilityTimestampDeletedText() {
+    Key keyBuilt =
+        Key.builder()
+            .row(rowText)
+            .family(familyText)
+            .qualifier(qualifierText)
+            .visibility(visibilityText)
+            .timestamp(10L)
+            .deleted(true)
+            .build();
+    Key keyExpected = new Key(rowText, familyText, qualifierText, visibilityText, 10L);
+    keyExpected.setDeleted(true);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowVisibilityText() {
+    Key keyBuilt = Key.builder().row(rowText).visibility(visibilityText).build();
+    Key keyExpected = new Key(rowText, new Text(), new Text(), visibilityText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowFamilyVisibilityText() {
+    Key keyBuilt = Key.builder().row(rowText).family(familyText).visibility(visibilityText).build();
+    Key keyExpected = new Key(rowText, familyText, new Text(), visibilityText);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingFromRowTimestampText() {
+    Key keyBuilt = Key.builder().row(rowText).timestamp(3L).build();
+    Key keyExpected = new Key(rowText);
+    keyExpected.setTimestamp(3L);
+    assertEquals(keyExpected, keyBuilt);
+  }
+
+  @Test
+  public void testKeyBuildingReusingBytes() {
+    byte[] reuse = new byte[] { 1, 2, 3 };
+    KeyBuilder.Build keyBuilder = Key.builder(false).row(reuse);
+    Key keyBuilt = keyBuilder.build();
+    assertEquals(reuse, keyBuilt.getRowBytes());
```

Comment:
Could use `assertSame()` here.  Its not needed for correctness in this case, just makes intentions clear.